tensorboard/uploader/server_info.py

```diff
@@ -152,6 +154,32 @@ def allowed_plugins(server_info):
         return frozenset((scalars_metadata.PLUGIN_NAME,))
 
 
+def max_blob_size(server_info):
+    """Determines the maximum allowed size for blob uploads.
+
+    This pulls from the `max_blob_size` on the `server_info` when that
+    submessage is set, else falls back to a default.
+
+    Note that the RPC endpoint also enforces a limit, which should be kept in
+    sync with the one provided by server_info.  Here the purpose is to provide
+    a better error message and to avoid attempting the upload if it is expected
+    to fail anyway.
+
+    Args:
+      server_info: A `server_info_pb2.ServerInfoResponse` message.
+
+    Returns:
+      A integer giving the maximum size allowed for blob uploads, in bytes.
+    """
+    if server_info.HasField("max_blob_size"):
```

Comment:
On the proto, `max_blob_size` is an `int64`, so this is equivalent to
just `if server_info.max_blob_size` (i.e., if-nonzero). IMHO this makes
the code a bit misleading, as it suggests that `max_blob_size` can be
“unset”. Can we either clarify that we’re testing the integer value of
the field or just `return si.max_blob_size or _DEFAULT`?

Reply:
Good point.  By adding the submessage, we do now have to test for the presence of `upload_limits`.  If it is present, then then`max_blob_size=0` will now be interpreted in the natural way (don't upload blobs).